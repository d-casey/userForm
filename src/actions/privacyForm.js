import {
  SUBMIT_PRIVACY_FORM,
  RESET_PRIVACY_FORM
} from '../constants/actions'

export const submitPrivacyForm = formFields => ({
  type: SUBMIT_PRIVACY_FORM,
  payload: formFields
})

export const resetPrivacyForm = () => ({
  type: RESET_PRIVACY_FORM
})
