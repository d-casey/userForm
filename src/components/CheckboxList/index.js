import React from 'react'
import CheckboxField from '../CheckboxField'
import PropTypes from 'prop-types'

const CheckboxList = ({ checkboxData, register }) => {
  const checkboxFields = checkboxData.map((checkbox) =>
    <CheckboxField register={register} key={checkbox.name} name={checkbox.name} defaultChecked={checkbox.defaultChecked} message={checkbox.message} testId={checkbox.testId} />
  )

  return (
    <ul className='full-width-imput'>
      {checkboxFields}
    </ul>
  )
}

CheckboxField.propTypes = {
  checkboxData: PropTypes.array,
  register: PropTypes.func.isRequired
}


export default CheckboxList
