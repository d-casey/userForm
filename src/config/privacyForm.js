const privacyForm = {
  checkboxData: [
    {
      name: 'receive_updates',
      defaultChecked: true,
      message: 'Receive Updates about Tray.io products by email',
      testId: 'receive_updates'
    },
    {
      name: 'receive_communication',
      message: 'Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team',
      testId: 'receive_communication'
    }
  ]
}

export default privacyForm
