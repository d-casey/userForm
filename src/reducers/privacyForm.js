import {
  SUBMIT_PRIVACY_FORM
} from '../constants/actions'

const initialState = {
  receive_updates: true,
  receive_communication: false,
  isSubmitted: false
}

export default function newUserForm(state = initialState, action) {
  switch (action.type) {
    case SUBMIT_PRIVACY_FORM:
    console.log('payload: ', action.payload)
      return {
        ...state,
        receive_updates: action.payload.receive_updates,
        receive_communication: action.payload.receive_communication,
        isSubmitted: true
      }

    default:
      return state
  }
}
