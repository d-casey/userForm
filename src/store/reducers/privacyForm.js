import {
  RESET_PRIVACY_FORM,
  SUBMIT_PRIVACY_FORM
} from '../constants/actions'

const initialState = {
  receive_updates: true,
  receive_communication: false,
  isSubmitted: false
}

export default (state = initialState, action) => {
  switch (action.type) {
    case SUBMIT_PRIVACY_FORM:
      return {
        ...state,
        receive_updates: action.payload.receive_updates,
        receive_communication: action.payload.receive_communication,
        isSubmitted: true
      }

      case RESET_PRIVACY_FORM:
        return initialState

    default:
      return state
  }
}
