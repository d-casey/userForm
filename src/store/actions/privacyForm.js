import {
  SUBMIT_PRIVACY_FORM,
  RESET_PRIVACY_FORM
} from '../constants/actions'

import { updateStage } from '../actions'

export const updatePrivacyFields = formFields => ({
  type: SUBMIT_PRIVACY_FORM,
  payload: formFields
})

export const submitPrivacyForm = formFields => dispatch => {
  dispatch(updatePrivacyFields(formFields))
  dispatch(updateStage(2))
}

export const resetPrivacyForm = () => ({
  type: RESET_PRIVACY_FORM
})
