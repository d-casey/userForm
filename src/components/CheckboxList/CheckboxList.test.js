import React from 'react'
import { create } from 'react-test-renderer'
import CheckboxList from '.'
import privacyForm from '../../config/privacyForm'

test('renders correctly when passed an array of checkbox data', () => {
  const register = jest.fn()
  const component = create(
    <CheckboxList
      checkboxData={privacyForm.checkboxData}
      register={register}
    />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
