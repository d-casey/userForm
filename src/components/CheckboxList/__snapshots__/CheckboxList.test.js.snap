// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when passed an array of checkbox data 1`] = `
<ul
  className="full-width-imput"
>
  <li
    className="checkbox"
  >
    <input
      className="checkbox-input"
      defaultChecked={true}
      name="receive_updates"
      type="checkbox"
    />
    <span>
      Receive Updates about Tray.io products by email
    </span>
  </li>
  <li
    className="checkbox"
  >
    <input
      className="checkbox-input"
      name="receive_communication"
      type="checkbox"
    />
    <span>
      Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team
    </span>
  </li>
</ul>
`;
